javascript/javascript3/week3/shopping.js

```diff
@@ -0,0 +1,93 @@
+class Product {
+    constructor(name, price) {
+        this.name = name;
+        this.price = price;
+    }
+    convertToCurrency(currency) {
+        if (currency.toLowerCase().includes('dollar')) {
+            return this.price * 0, 16
+        } else if (currency.toLowerCase().includes('euro')) {
+            return this.price * 0, 13
+        } else if (currency.toLowerCase().includes('pound')) {
+            return this.price * 0, 12
+        } else {
+            return this.price
+        }
+    }
+}
+
+class ShoppingCart {
+    constructor() {
+        this.products = [];
+    }
+
+    addProduct(product) {
+        this.products.push(product)
+    }
+
+    removeProduct(product) {
+        const updatedProducts = this.products.filter((it) => product.name !== it.name)
+        this.products = updatedProducts
+    }
+
+    searchProduct(productName) {
+        return this.products.filter(a => a.name === productName)
+    }
+
+    getTotal() {
+        return this.products.reduce((a, b) => a.price + b.price)
+    }
```

Comment:
This function does not work well, it returns `NaN` beacuse the `reduce` method is used correctly.

You can fix it by for example:

```suggestion
    getTotal() {
        return this.products.reduce((a, b) => a + b.price, 0)
    }
```

Reply:
strange.. but i got output on screen .. but thanks